tests/test_odxtools.py

```diff
@@ -12,7 +12,7 @@
 
 # use the diag layer container's document fragments as the default for
 # resolving references
-container_doc_frags = odxdb.diag_layer_containers.somersault.odx_id.doc_fragments
+container_doc_frags = odxdb.diag_layer_containers["somersault"].odx_id.doc_fragments
```

Comment:
If you can't do `odxdb.diag_layer_containers.somersault.odx_id.doc_fragments` then `NamedItemList` type hinting is not correct

Reply:
the problem is that these attributes are all dynamically added at runtime (by extending `__dict__` of the named item list) and there is no way that mypy can be aware of this. (I also cannot come up with a "clean" way to tell `mypy` short of modifying the `mypy` source code...)

Reply:
according to mypy issue tracker, it is supported. 
https://github.com/python/mypy/pull/544
https://github.com/python/mypy/issues/520

Reply:
interesting, I didn't know about `__getattr__()` (it still feels a bit like voodoo to me). Anyway, I've adapted the implementation of `ItemList`: 17221463c280b9d8d95201fd14c0f173c415a513 (this is a drive-by patch but I hope this is okay for you in this instance...)